### ISSUE TITLE
Docs hosted on readthedocs.io are missing content

### ISSUE THREAD

**Issue by birdcolour** [spotify/luigi#2464]
I've noticed that a few pages in the contribs documentation appear not to have been built properly, and are missing all of the actual content:

http://luigi.readthedocs.io/en/stable/api/luigi.contrib.hadoop.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.hadoop_jar.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.hive.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.opener.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.redshift.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.s3.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.scalding.html
http://luigi.readthedocs.io/en/stable/api/luigi.contrib.sge.html

These are all of the pages in `luigi/contrib`, but I have no idea if this is isolated to just here. The docstrings are definitely there.

**Comment by dlstadther** [comment]
Interesting... Thanks for raising @birdcolour 

@Tarrasch , i have no experience with our readthedocs sphinx implementation. Do you have any ideas or recommendations here?

**Comment by dlstadther** [comment]
Actually, looks like @jcrobak did some of the sphinx setup. Maybe he can help too

**Comment by Tarrasch** [comment]
Hmm... Feel free to take a look @birdcolour, I unfortunately don't have the time to dive in.

**Comment by jcrobak** [comment]
It looks like the "API Documentation" section, which describes the steps required for new packages didn't make it through a migration 5899a3d231a8f59216e370f4bd552aef43e34e88

tl;dr: `pip install sphinx && sphinx-apidoc -o doc/api -T luigi` will rebuild`.

With that said, `tox -e docs` is failing for me related to https://github.com/sphinx-doc/sphinx/issues/2443

HTH!

**Comment by lizmeister321** [comment]
Hey, any update on this fix?

**Comment by dlstadther** [comment]
If i build docs locally, i get all the correct api html files

**Comment by jethron** [comment]
Might be related: the `sphinx-build` command from `tox.ini` fails in `luigi/configuration/cfg_parser.py` because `LuigiConfigParser.get` has no docstring, and so Sphinx takes the inherited docstring from the stdlib `ConfigParser.get` which contains odd unbalanced quotes (open with \` and close with ') that trigger a warning in Sphinx's parsing. Since the build has warnings as errors enabled the docs build aborts. Same thing happens if you try to override `has_option` or another method of `ConfigParser` that uses the same quote convention without supplying a new docstring without that convention.

If you add a docstring (empty or otherwise) to `LuigiConfigParser.get` to avoid the inheritance the build succeeds. Not sure if it's a configparser bug because the quote convention is weird, a Sphinx bug because it can't parse docstrings in the stdlib, or something that should just be fixed in this project to avoid inheriting this specific docstring.

The docstring value in question on my system, for reference:
> "Get an option value for a given section.\n\n        If \`vars' is provided, it must be a dictionary. The option is looked up\n        in \`vars' (if provided), \`section', and in \`DEFAULTSECT' in that order.\n        If the key is not found and \`fallback' is provided, it is used as\n        a fallback value. \`None' can be provided as a \`fallback' value.\n\n        If interpolation is enabled and the optional argument \`raw' is False,\n        all interpolations are expanded in the return values.\n\n        Arguments \`raw', \`vars', and \`fallback' are keyword only.\n\n        The section DEFAULT is special.\n        "

And the build error:
> Warning, treated as error: 
> docstring of luigi.configuration.LuigiConfigParser.get:3:Inline interpreted text or phrase reference start-string without end-string.

**Comment by jethron** [comment]
Ah, sorry @dlstadther, that only applies if you try to build the docs in Python 3, and RTD is building in Py2 so it's a different problem.

Going through the build log, to fix these pages it looks like it's a simple case of boto3 not being installed in the RTD environment.

Adding `install_requires.append('boto3')` to `setup.py` alongside sqlalchemy in the RTD check should fix this properly for the RTD build on Python2.7.

**Comment by dlstadther** [comment]
Hey @jethron Could you link me to the build log you're referring to?

**Comment by jethron** [comment]
https://readthedocs.org/projects/luigi/builds/7832345/

Is one of the latest builds that passed, it has started failing since some of the latest merges. At the bottom of the 9th step you see a bunch of import failures for botocore.

**Comment by Tarrasch** [comment]
Ah yes I recall there's some silent failures happening on that site. I think we're on the right track.

**Comment by dlstadther** [comment]
Docs are back in business!
